js/common.js

```diff
@@ -3,17 +3,25 @@ if (!window.console) {
   window.console = {log: Prototype.emptyFunction};
 }
 
-// Implement onhashchange support
+if (!window.location.moz_queryString) {
+  window.location.moz_queryString = function() {
+    var qs = window.location.href;
+    var i = qs.indexOf("?");
+    return i == -1 ? "" : qs.substring(i);
+  }
+}
+
+// Poll for changes to the query string
 (function() {
-  var hash = window.location.hash;
+  var qs = window.location.moz_queryString();
```

Comment:
I know the code isn't great to begin with, but can you use full variable names so this is easy to read? `queryString` is easier to scan than `qs`.